### ISSUE TITLE
Bug: Trailer/ Bale weights quick form not submitting

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#186]
HI @mstenta, @paul121 

The team at North Wyke are having a problem submitting the Harvest/ bale weights quick form. All the fields that I can see as required are filled in, but we get the below error code.

The details so you can try recreate the error are:

Location: Twittleham
Tractor: John Deere 6620 (Big Boots)
Harvest Eqpt: Round bailer
Type: Bailing
Total Number bales: Wrapped bales 15 round bales **(this was the bit it didn't seem to like, highlighted in red)**
Operation: 06 May 09:30 for 45 minutes
Tractor hours: 7664.5 to 7665.3
Storage location: SRF Compound
Operator: Jon
Status: Done

I had a go with the same data and got the same error. But let me know if I am overlooking something.

Thanks so much,

Aislinn

![image](https://user-images.githubusercontent.com/16349781/167897466-67f9f670-e8f7-4a26-9faf-51c634270f5c.png)

**Comment by mstenta** [comment]
@aislinnpearson Hmm that is a common Drupal error that sometimes happens when the "form state" becomes outdated. When you tested, was it in a new browser tab or a different computer, perchance? Or the same one that failed originally? Trying it in a new browser tab *might* resolve it...

If not, then it may be related to version 2.1.0, which I deployed earlier today.

**Comment by paul121** [comment]
I saw a similar issue after changing the form validation for select fields whose options are  populated via AJAX. I fixed that here: https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/commit/338b2e142a0f98c64b1508b67b655796f37221e6

I think the issue is something similar to this. Except the bale weight units aren't populated  via AJAX.. hmm.

**Comment by aislinnpearson** [comment]
Hi @mstenta 

> When you tested, was it in a new browser tab or a different computer, perchance? Or the same one that failed originally? Trying it in a new browser tab might resolve it...

I tried entering it on a completely separate computer in a newly opened tab.

I've shown Josh the link, so they can follow the conversation here now. :)

**Comment by paul121** [comment]
Spoke with @aislinnpearson just now because I couldn't reproduce this issue. We discovered this is being caused because the `Grass/Straw Bale Types` parent term did not exist in the `unit` taxonomy. The `farm_rothasmted_dev` module creates this in my dev environment. This is required for populating the bale type options in the trailer harvest quick form. Since no options were rendered we get the "Illegal choice" error message.

Originally I had considered adding some code to alert when these taxonomy hierarchies were not configured. I chose not to do this because it would have been more work and raises the question what is the most effective way to "alert" when there is an issue, prevent form submission, etc... I think we should work on documenting these little steps that are required!
